supervisor/addons/configuration.py

```diff
@@ -0,0 +1,14 @@
+"""Confgiuration Objects for Addon Config."""
```

Comment:
let's rename `addons/configuration.py` to `addons/data.py` like we use on other place

Reply:
Looks like `addons/data.py` already exists and it depends on `store.addon.AddonStore` which depends on `addons.model.AddonModel` which is where we need our `FolderMapping` dataclass. Moving `FolderMapping` to data.py creates a circular reference:

`ImportError: cannot import name 'AddonStore' from partially initialized module 'supervisor.store.addon' (most likely due to a circular import) (/usr/src/supervisor/supervisor/store/addon.py)`

Is there somewhere better for the `FolderMapping` dataclass or does `addons/data.py` need to be refactored? I left it in `configuration.py` for now.

Reply:
Yeah, you have to rename the `__init__.py` inside `addons` to `manager.py` and create an empty new `__init__.py`. We did that already for some module but only if it was needed because of the shifting from const.py

Reply:
I'm so sorry, I'm new to this repo so please bear with me. Should the `FolderMapping` dataclass be moved into the new `addons/manager.py`? Or should the logic in `addons/data.py` be moved to `addons/manager.py` and then `FolderMapping` go into `addons/data.py`?

Thanks for you help!

Reply:
FolderMapping goes into data.py. Essentially you just rename `addons/__init__.py` to `addons/manager.py` and then make a new `addons/__init__.py` that only has a docstring. It's a small change but can be annoying since you have to refactor all existing references to it (well VSCode does most of the work but patches in test sometimes need manual fixing).

To keep it simple, I made a PR just for that. @pvizeli let's get #4760 merged so @jgoakley can rebase off it and not have the circular import issue.

Reply:
Thanks @mdegat01 for creating that pull request! After it was merged I updated my branch to include those changes but I'm still getting the circular reference error when moving the `FolderMapping` dataclass to `addons/data.py`

I made the change anyway and pushed it to the branch so everyone can see the issue. I may be adding the dataclass incorrectly so please let me know if there is anything I should do differently. Thanks!